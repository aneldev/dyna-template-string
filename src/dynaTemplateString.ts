export const dynaTemplateString = (
  {
    text,
    variables,
  }
    : {
    text: string;
    variables: {
      [variableName: string]: string;
    };
  },
): string => {
  let output = text;
  Object.keys(variables)
    .forEach(variable =>
      output = output
        .replace(
          new RegExp('\\${' + variable + '}', 'g'),
          variables[variable],
        )
    );
  return output;
};
