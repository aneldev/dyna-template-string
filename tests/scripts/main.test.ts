import {dynaTemplateString} from "../../src"

describe('dynaTemplateString', () => {
  it('A Variable', () => {
    expect(
      dynaTemplateString({
        text: 'Hello ${name}! How are you ${name} today?',
        variables: {
          name: 'John',
        },
      })
    )
      .toMatchSnapshot();
	});
  it('Variables and line breaks', () => {
    expect(
      dynaTemplateString({
        text: 'Hello ${salutation} ${lastName}!\nWill be a long day ${firstName}. You won\'t forget your name ${firstName}!',
        variables: {
          salutation: 'Mr',
          firstName: 'John',
          lastName: 'Smith',
        },
      })
    )
      .toMatchSnapshot();
	});
  it('Undefined variable', () => {
    expect(
      dynaTemplateString({
        text: 'Hello ${salutation} ${lastName}!\nThe report will be ready at ${date}',
        variables: {
          salutation: 'Mr',
          lastName: 'Smith',
        },
      })
    )
      .toMatchSnapshot();
	});
});
