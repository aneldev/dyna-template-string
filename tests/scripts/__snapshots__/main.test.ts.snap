// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynaTemplateString A Variable 1`] = `"Hello John! How are you John today?"`;

exports[`dynaTemplateString Undefined variable 1`] = `
"Hello Mr Smith!
The report will be ready at \${date}"
`;

exports[`dynaTemplateString Variables and line breaks 1`] = `
"Hello Mr Smith!
Will be a long day John. You won't forget your name John!"
`;
